File: platform-kotlin-mobile/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmobile/parser/KotlinCommonPlatformConfigParser.java
package ru.pocketbyte.locolaser.platform.kotlinmobile.parser;

import ru.pocketbyte.locolaser.config.platform.BasePlatformConfig;
import ru.pocketbyte.locolaser.exception.InvalidConfigException;
import ru.pocketbyte.locolaser.platform.kotlinmobile.KotlinCommonPlatformConfig;
import ru.pocketbyte.locolaser.platform.mobile.parser.BaseMobilePlatformConfigParser;

public class KotlinCommonPlatformConfigParser extends BaseMobilePlatformConfigParser {

    @Override
    protected BasePlatformConfig platformByType(String type, boolean throwIfWrongType) throws InvalidConfigException {
        if (KotlinCommonPlatformConfig.TYPE.equals(type))
            return new KotlinCommonPlatformConfig();

        if (throwIfWrongType)
            throw new InvalidConfigException("Unknown platform: " + type);

        return null;
    }
}

File: platform-kotlin-mobile/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmobile/parser/KotlinCommonPlatformConfigParser.kt
package ru.pocketbyte.locolaser.platform.kotlinmobile.parser

import ru.pocketbyte.locolaser.config.platform.BasePlatformConfig
import ru.pocketbyte.locolaser.exception.InvalidConfigException
import ru.pocketbyte.locolaser.platform.kotlinmobile.KotlinCommonPlatformConfig
import ru.pocketbyte.locolaser.platform.mobile.parser.BaseMobilePlatformConfigParser

class KotlinCommonPlatformConfigParser : BaseMobilePlatformConfigParser() {

    @Throws(InvalidConfigException::class)
    override fun platformByType(type: String?, throwIfWrongType: Boolean): BasePlatformConfig? {
        if (KotlinCommonPlatformConfig.TYPE == type)
            return KotlinCommonPlatformConfig()

        if (throwIfWrongType)
            throw InvalidConfigException("Unknown platform: $type")

        return null
    }
}
